feat(http): add support for fetch referrerPolicy option in HttpClient

SkyZeroZx opened the pull request:
## Motivation  

- Controlling how much referrer information is included with requests.  
- Enabling developers to explicitly set stricter privacy and security rules.  
- Improving compliance with server policies and modern web platform defaults.  


## Examples of New Usage

```ts
 
http.get('/api/user-data', {
  referrerPolicy: 'strict-origin-when-cross-origin'
}).subscribe();

```

JeanMeche reviewed (approved):
reviewed-for: public-api

SkyZeroZx committed:
feat(http): add support for fetch referrerPolicy option in HttpClient

Introduces a referrer policy option for HTTP requests to allow specifying the referrer information sent, improving privacy and security controls.

mmalerba commented:
This PR was merged into the repository. The changes were merged into the following branches:

- main: 5cbdefcf118e9c228bc887be30114afc84a4db2a
